command/crl/signature_algorithms.go

```diff
@@ -79,82 +75,18 @@ func (s SignatureAlgorithm) String() string {
-type pssParameters struct {
-	// The following three fields are not marked as
-	// optional because the default values specify SHA-1,
-	// which is no longer suitable for use in signatures.
-	Hash         pkix.AlgorithmIdentifier `asn1:"explicit,tag:0"`
-	MGF          pkix.AlgorithmIdentifier `asn1:"explicit,tag:1"`
-	SaltLength   int                      `asn1:"explicit,tag:2"`
-	TrailerField int                      `asn1:"optional,explicit,tag:3,default:1"`
-}
-
-func newSignatureAlgorithm(ai pkix.AlgorithmIdentifier) SignatureAlgorithm {
+func newSignatureAlgorithm(xsa x509.SignatureAlgorithm) (SignatureAlgorithm, error) {
 	sa := SignatureAlgorithm{
-		OID: ai.Algorithm.String(),
-	}
-
-	if ai.Algorithm.Equal(oidSignatureEd25519) {
-		// RFC 8410, Section 3
-		// > For all of the OIDs, the parameters MUST be absent.
-		if len(ai.Parameters.FullBytes) != 0 {
-			return sa
-		}
-	}
-
-	if !ai.Algorithm.Equal(oidSignatureRSAPSS) {
-		for _, details := range signatureAlgorithmDetails {
-			if ai.Algorithm.Equal(details.oid) {
-				sa.Name = details.algo.String()
-				sa.algo = details.algo
-				sa.hash = details.hash
-			}
-		}
-		return sa
-	}
-
-	// RSA PSS is special because it encodes important parameters
-	// in the Parameters.
-
-	var params pssParameters
-	if _, err := asn1.Unmarshal(ai.Parameters.FullBytes, &params); err != nil {
-		return sa
-	}
-
-	var mgf1HashFunc pkix.AlgorithmIdentifier
-	if _, err := asn1.Unmarshal(params.MGF.Parameters.FullBytes, &mgf1HashFunc); err != nil {
-		return sa
-	}
-
-	// PSS is greatly overburdened with options. This code forces them into
-	// three buckets by requiring that the MGF1 hash function always match the
-	// message hash function (as recommended in RFC 3447, Section 8.1), that the
-	// salt length matches the hash length, and that the trailer field has the
-	// default value.
-	if (len(params.Hash.Parameters.FullBytes) != 0 && !bytes.Equal(params.Hash.Parameters.FullBytes, asn1.NullBytes)) ||
-		!params.MGF.Algorithm.Equal(oidMGF1) ||
-		!mgf1HashFunc.Algorithm.Equal(params.Hash.Algorithm) ||
-		(len(mgf1HashFunc.Parameters.FullBytes) != 0 && !bytes.Equal(mgf1HashFunc.Parameters.FullBytes, asn1.NullBytes)) ||
-		params.TrailerField != 1 {
-		return sa
+		Name: xsa.String(),
+		algo: xsa,
 	}
 
-	switch {
-	case params.Hash.Algorithm.Equal(oidSHA256) && params.SaltLength == 32:
-		sa.Name = x509.SHA256WithRSAPSS.String()
-		sa.algo = x509.SHA256WithRSAPSS
-		sa.hash = crypto.SHA256
-	case params.Hash.Algorithm.Equal(oidSHA384) && params.SaltLength == 48:
-		sa.Name = x509.SHA384WithRSAPSS.String()
-		sa.algo = x509.SHA384WithRSAPSS
-		sa.hash = crypto.SHA384
-	case params.Hash.Algorithm.Equal(oidSHA512) && params.SaltLength == 64:
-		sa.Name = x509.SHA512WithRSAPSS.String()
-		sa.algo = x509.SHA512WithRSAPSS
-		sa.hash = crypto.SHA512
+	sad, ok := signatureAlgorithmMap[xsa]
+	if !ok {
+		return sa, fmt.Errorf("unknown or unsupported signature algorithm: %s", sa.Name)
 	}
```

Comment:
We should not fail, if Go is not able to identify the algorithm it the `x509.SignatureAlgorithm` will be `x509.UnknownSignatureAlgorithm` we should be able to display in text and in json something like:
```
Signature Algorithm: UNKNOWN
```

```json
"signature": {
    "signature_algorithm": {
      "name": "unknown",
      "oid": "unknown"
    },
    "value": "...",
    "valid": false
}